Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs

```diff
@@ -282,30 +282,24 @@ public async Task<HttpResponseInfo> SendAsync(HttpRequestOptions options, string
             return response;
         }
 
-        private async Task<HttpResponseInfo> GetCachedResponse(string responseCachePath, TimeSpan cacheLength, string url)
+        private HttpResponseInfo GetCachedResponse(string responseCachePath, TimeSpan cacheLength, string url)
         {
             try
             {
                 if (_fileSystem.GetLastWriteTimeUtc(responseCachePath).Add(cacheLength) > DateTime.UtcNow)
                 {
-                    using (var stream = _fileSystem.GetFileStream(responseCachePath, FileOpenMode.Open, FileAccessMode.Read, FileShareMode.Read, true))
-                    {
-                        var memoryStream = new MemoryStream();
-
-                        await stream.CopyToAsync(memoryStream).ConfigureAwait(false);
-                        memoryStream.Position = 0;
+                    var stream = _fileSystem.GetFileStream(responseCachePath, FileOpenMode.Open, FileAccessMode.Read, FileShareMode.Read, true);
 
-                        return new HttpResponseInfo
-                        {
-                            ResponseUrl = url,
-                            Content = memoryStream,
-                            StatusCode = HttpStatusCode.OK,
-                            ContentLength = memoryStream.Length
-                        };
-                    }
+                    return new HttpResponseInfo
+                    {
+                        ResponseUrl = url,
+                        Content = stream,
+                        StatusCode = HttpStatusCode.OK,
+                        ContentLength = stream.Length
+                    };
                 }
             }
-            catch (FileNotFoundException)
+            catch (FileNotFoundException) // REVIEW: @bond Is this really faster?
```

Comment:
I don't really get why it should be faster. Either way, you're going to need to wait for I/O. I would expect performance to be almost equal.

Reply:
That's what benchmarks are for :p